types/encode.go

```diff
@@ -103,16 +103,33 @@ func NIPSTChallengeAsBytes(challenge *NIPSTChallenge) ([]byte, error) {
 	return w.Bytes(), nil
 }
 
-func TransactionAsBytes(tx *SerializableTransaction) ([]byte, error) {
+func AddresableTransactionAsBytes(tx *AddressableSignedTransaction) ([]byte, error) {
 	var w bytes.Buffer
 	if _, err := xdr.Marshal(&w, &tx); err != nil {
 		return nil, fmt.Errorf("error marshalling transaction: %v", err)
 	}
 	return w.Bytes(), nil
 }
 
-func BytesAsTransaction(buf []byte) (*SerializableTransaction, error) {
-	b := SerializableTransaction{}
+func BytesAsAddressableTransaction(buf []byte) (*AddressableSignedTransaction, error) {
```

Comment:
you can replace calls to these by using BytesToInterface and interfaceToBytes methods

Reply:
We can discuss it offline. I prefer type safety over minor code dup.